### PR TITLE
[Feature] Support get flops tool

### PR DESCRIPTION
**Pull request by liuyanyi**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Issue #145 
add `get_flops` from https://github.com/open-mmlab/mmdetection/blob/master/tools/analysis_tools/get_flops.py
Add `docs/en/useful_tools.md`

## Modification
Add `tools/analysis_tools/get_flops.py`
Add forward_dummy for `oriented_rcnn` and `oriented_standard_roi_head`

S2ANet still has some issues with flops, because mmcv doesn't support dcn and orconv flops.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by zytx121** [comment]
Good job! Maybe we should add a document in `get_flops` to tell users which models we support now.

**Comment by liuyanyi** [comment]
> Good job! Maybe we should add a document in `get_flops.py` to tell users which models we support now.

Sure, I'll add a doc like [useful_tools](https://github.com/open-mmlab/mmdetection/blob/master/docs/en/useful_tools.md) in mmdet.

**Comment by yangxue0827** [comment]
`docs/zh_cn` also need to be updated.

File: docs/en/useful_tools.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+ filename.
+
+```shell
+python tools/model_converters/publish_model.py ${INPUT_FILENAME} ${OUTPUT_FILENAME}
+```
+
+E.g.,
+
+```shell
+python tools/model_converters/publish_model.py work_dirs/faster_rcnn/latest.pth faster_rcnn_r50_fpn_1x_20190801.pth
```
Comment (zytx121) [inline docs/en/useful_tools.md:109]: Maybe we should change `faster_rcnn` to `rotated_faster_rcnn`.

faster_rcnn_r50_fpn_1x_20190801.pth -> rotated_faster_rcnn_r50_fpn_1x_dota_le90-0393aa5c.pth

File: docs/en/useful_tools.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+```
+
+The final output filename will be `faster_rcnn_r50_fpn_1x_20190801-{hash id}.pth`.
```
Comment (zytx121) [inline docs/en/useful_tools.md:112]: faster_rcnn_r50_fpn_1x_20190801-{hash id}.pth -> rotated_faster_rcnn_r50_fpn_1x_dota_le90_20190801-{hash id}.pth

File: docs/zh_cn/useful_tools.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
+除了训练和测试脚本，我们在`tools/`文件夹内还提供了一些有用的工具。
+
+## 日志分析
+
+`tools/analysis_tools/analyze_logs.py` 通过给定的日志文件绘制loss/mAP曲线。 需首先执行`pip install seaborn`安装依赖。
```
Comment (zytx121) [inline docs/zh_cn/useful_tools.md:5]: 中文和英文之间需要空格：

绘制 loss/mAP 曲线
执行 `pip install seaborn` 安装

File: docs/zh_cn/useful_tools.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+![loss curve image](https://raw.githubusercontent.com/open-mmlab/mmdetection/master/resources/loss_curve.png)
+
+示例:
+
+- 绘制某次执行的分类损失
+
+    ```shell
+    python tools/analysis_tools/analyze_logs.py plot_curve log.json --keys loss_cls --legend loss_cls
+    ```
+- 绘制某次执行的分类和回归损失，同时将图像保存到pdf文件
```
Comment (zytx121) [inline docs/zh_cn/useful_tools.md:20]: 保存到 pdf 文件

File: docs/zh_cn/useful_tools.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+    ```shell
+    python tools/analysis_tools/analyze_logs.py plot_curve log.json --keys loss_cls loss_bbox --out losses.pdf
+    ```
+
+- 在同一张图像中比较两次执行的mAP
```
Comment (zytx121) [inline docs/zh_cn/useful_tools.md:26]: 执行的 mAP

File: docs/zh_cn/useful_tools.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+`tools/misc/browse_dataset.py` 帮助用户浏览检测的数据集（包括图像和检测框的标注），或将图像保存到指定目录。
+
+```shell
+python tools/misc/browse_dataset.py ${CONFIG} [-h] [--skip-type ${SKIP_TYPE[SKIP_TYPE...]}] [--output-dir ${OUTPUT_DIR}] [--not-show] [--show-interval ${SHOW_INTERVAL}]
+```
+
+## 模型复杂度
+
+`tools/analysis_tools/get_flops.py` 是改编自 [flops-counter.pytorch](https://github.com/sovrasov/flops-counter.pytorch) 的脚本，用于计算给定模型的FLOPs和参数量.
```
Comment (zytx121) [inline docs/zh_cn/useful_tools.md:60]: 模型的 FLOPs 和参数量

File: docs/zh_cn/useful_tools.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+Flops: 215.92 GFLOPs
+Params: 36.42 M
+==============================
+```
+
+**注意**: 此工具仍处于实验阶段，我们并不能保证计算结果是绝对正确的。你可以将结果用于简单的比较，
+但在技术报告或论文中采用之前请仔细检查
+
+1. FLOPs 与输入大小相关，但参数量与其无关。默认输入大小是(1, 3, 1024, 1024).
+2. 一些算子例如DCN或自定义算子并未包含在FLOPs计算中，所以S<sup>2</sup>A-Net和基于RepPoints的模型的FLOPs计算是错误的。
```
Comment (zytx121) [inline docs/zh_cn/useful_tools.md:80]: 一些算子例如 DCN 或自定义算子并未包含在 FLOPs 计算中，所以 S<sup>2</sup>A-Net 和基于 RepPoints 的模型的 FLOPs 计算是错误的。

File: docs/zh_cn/useful_tools.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+python tools/model_converters/publish_model.py work_dirs/faster_rcnn/latest.pth faster_rcnn_r50_fpn_1x_20190801.pth
+```
+
+最终输出的文件名是 `faster_rcnn_r50_fpn_1x_20190801-{hash id}.pth`.
+
+## 基准测试
+
+### FPS基准
+
+`tools/analysis_tools/benchmark.py` 帮助用户计算FPS。 FPS值包括模型前向传播和后处理。为了得到更准确的数值，目前只支持单GPU分布式启动。
```
Comment (zytx121) [inline docs/zh_cn/useful_tools.md:110]: 帮助用户计算 FPS。 FPS 值包括模型前向传播和后处理。为了得到更准确的数值，目前只支持单 GPU 分布式启动。

File: docs/zh_cn/useful_tools.md [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+详细信息请查看 [`mmcv.cnn.get_model_complexity_info()`](https://github.com/open-mmlab/mmcv/blob/master/mmcv/cnn/utils/flops_counter.py)。
+3. 两阶段检测器的 FLOPs 取决于候选的数量。
+
+### 准备发布模型
+
+`tools/model_converters/publish_model.py` 帮助用户准备他们将发布的模型。
+
+在将模型上传到 AWS 之前，你可能需要
+
+1. 将模型权重转换至CPU
```
Comment (zytx121) [inline docs/zh_cn/useful_tools.md:90]: 将模型权重转换至 CPU

File: docs/zh_cn/useful_tools.md [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+3. 计算权重文件的哈希值并附加到文件名后
+
+```shell
+python tools/model_converters/publish_model.py ${INPUT_FILENAME} ${OUTPUT_FILENAME}
+```
+
+例如,
+
+```shell
+python tools/model_converters/publish_model.py work_dirs/faster_rcnn/latest.pth faster_rcnn_r50_fpn_1x_20190801.pth
```
Comment (zytx121) [inline docs/zh_cn/useful_tools.md:101]: Maybe we should change faster_rcnn to rotated_faster_rcnn.

**Review by zytx121 — APPROVED** [review]
LGTM

**Review by ZwwWayne — APPROVED** [review]

**Comment by hahakid** [comment]
warning that the path of `docs/en/useful_tools.md' is wrongly inserted as "docs/en/tutorials/useful_tools.md".
Please refer line 108 of readme.
Same as the Chinese version (line 102).

**Pull request merged**